roof_classify/roof_classify.py

```diff
@@ -375,7 +389,28 @@ def getNumberofClasses(self):
         """
         roofingShapefileDir = self.dlg.lineEdit_2.text()
         shpFolder = Path(roofingShapefileDir).rglob("*.shp")
-        return len(shpFolder)
+        files = [shp for shp in shpFolder]
+        return len(files)
+
+    def convertRaster2Array(rasterFilepath):
+        """Convert a multiband raster into a numpy array.
+        Given a 3-band raster image of size ncols*nrows pixels, the function will return
+        an numpy array of shape (3, ncols, nrows).
+
+        :param rasterFilepath: Filepath of the raster to convert
+        :type rasterFilepath: str
+        :return: Converted image into array
+        :rtype: numpy.array
+        """
+        raster_dataset = gdal.Open(rasterFilepath, gdal.GA_ReadOnly)
+        bands_data = []
+        # Parsing the raster's bands
+        for b in range(1, raster_dataset.RasterCount + 1):
+            band = raster_dataset.GetRasterBand(b)
+            bands_data.append(band.ReadAsArray())
+        # Create an array of shape (nrows, ncols, ndim)
+        bands_data = np.dstack(bands_data)
+        return bands_data
 
     def rasterizeRoofingLayer(vectorFilepath, rasterFilepath, classNumber):
```

Comment:
Add params and returned types like:

```python
def myfunc(self, param1: str) -> int:
    return int(param1)
```

Reply:
Do you mean I should define local functions (by adding the `self` key word parameter) or you'd rather write `return np.dstack(bands_data)` and suppress the line `bands_data = np.dstack(bands_data)`?

Reply:
No, I just talk about type annotation which applies both to functions or methods.